files/scripts/swss.sh

```diff
@@ -97,6 +97,7 @@ start() {
         /usr/bin/docker exec database redis-cli -n 2 FLUSHDB
         /usr/bin/docker exec database redis-cli -n 5 FLUSHDB
         clean_up_tables 6 "'PORT_TABLE*', 'MGMT_PORT_TABLE*', 'VLAN_TABLE*', 'VLAN_MEMBER_TABLE*', 'INTERFACE_TABLE*', 'MIRROR_SESSION*', 'VRF_TABLE*'"
+        /usr/bin/docker exec database redis-cli -n 0 HSET "SWSS_DB_FLUSH_DONE" "1"
```

Comment:
One problem I foresee here is that `SWSS_DB_FLUSH_DONE` never gets set to 0. Take for instance the new logic I implemented in the 201803 branch (and will soon implement in the master/201811 branches, as well). If a critical process crashes in the swss container, the container will exit causing the swss service to restart itself and its dependent services. At this point, swss restarts and subsequently restarts teamd. It's _possible_ that teamd would check the value before the databases get flushed, yet `SWSS_DB_FLUSH_DONE` would still be set to 1, causing teamd to start before the databases are flushed. How can we reliably set this value to 0 when applicable?

Reply:
Yes, at least SWSS_DB_FLUSH_DONE should be set as 0 upon swss service stop.

Reply:
Did you test this change with system warm reboot?

Reply:
@yxieca I tested with cold reboot and teamd docker warm restart, but not system warm reboot.  Did you see any issue?

Reply:
Not  really.

I my test. I think you change is equal to add service teamd after syncd. I don't see behavior change from one to another. Can you test that change?

diff --git a/files/build_templates/teamd.service.j2 b/files/build_templates/teamd.service.j2
index 792b824..bde55c6 100644
--- a/files/build_templates/teamd.service.j2
+++ b/files/build_templates/teamd.service.j2
@@ -2,6 +2,7 @@
 Description=TEAMD container
 Requires=updategraph.service
 After=updategraph.service
+After=syncd.service
 Before=ntp-config.service
 
 [Service]

Reply:
Yes, with recent systemd service change, this should work too. 

The difference might be bootup speed (for both cold boot and warm boot), but I don't have any concrete data.